languages/thingtalk/en/constants.genie

```diff
@@ -44,21 +44,32 @@ constant_date_point = {
     'now'                 => C.makeDate(null, '+', null);
     'today'               => C.makeDate(new Ast.DateEdge('start_of', 'day'), '+', null);
     'yesterday'           => C.makeDate(new Ast.DateEdge('start_of', 'day'), '-', new Ast.Value.Measure(1, 'day'));
-    'this week'           => C.makeDate(new Ast.DateEdge('start_of', 'week'), '+', null);
 
     !turking {
-        'last week'           => C.makeDate(new Ast.DateEdge('start_of', 'week'), '-', new Ast.Value.Measure(1, 'week'));
-        'last month'          => C.makeDate(new Ast.DateEdge('end_of', 'mon'), '-', new Ast.Value.Measure(1, 'mon'));
-        'last year'           => C.makeDate(new Ast.DateEdge('end_of', 'year'), '-', new Ast.Value.Measure(1, 'year'));
-
         ?future_dates {
             'tomorrow'            => C.makeDate(new Ast.DateEdge('end_of', 'day'), '+', null);
             'the end of the day'  => C.makeDate(new Ast.DateEdge('end_of', 'day'), '+', null);
             'the end of the week' => C.makeDate(new Ast.DateEdge('end_of', 'week'), '+', null);
-            'this month'          => C.makeDate(new Ast.DateEdge('start_of', 'mon'), '+', null);
-            'this year'           => C.makeDate(new Ast.DateEdge('start_of', 'year'), '+', null);
-            'next month'          => C.makeDate(new Ast.DateEdge('end_of', 'mon'), '+', null);
-            'next year'           => C.makeDate(new Ast.DateEdge('end_of', 'year'), '+', null);
+            'the end of the month'=> C.makeDate(new Ast.DateEdge('end_of', 'mon'), '+', null);
+            'the end of the year' => C.makeDate(new Ast.DateEdge('end_of', 'year'), '+', null);
+        }
+    }
+}
+
+constant_date_range = {
+    'this week'   => [C.makeDate(new Ast.DateEdge('start_of', 'week'), '+', null), C.makeDate(new Ast.DateEdge('end_of', 'week'), '+', null)];
```

Comment:
Does this mean we annotate "news published this week" with two comparisons?

Reply:
yes, e.g.,  "show me meetings this week" (but in this case we want range between now and end of week, instead of start of week and end of week.....)

Reply:
That's a tricky one. We'll need a lot of annotations.